accepted/future-releases/small-features-21Q1/implementation-plan.md

```diff
@@ -0,0 +1,153 @@
+
+The analyzer also needs to support `const Symbol(">>>")`, but otherwise seems to support the `>>>` feature (`operator>>>`  and `#>>>` works).
+
+Otherwise the analyzer is unlikely to need significant change for any of the features.
+
+## Phase 2 (Tool Implementation)
+
+### dart2js
+
+If the feature is handled by the front end, there may be no dart2js work.
+Otherwise, dart2js may need to handle any Kernel changes 
+
+Dart2js supports `>>>` except for `new Symbol(">>>")`. This can be fixed by a single RegExp update.
+
+### Dartfmt
+
+The only new syntax is metadata type arguments. They need to be supported, but should likely be treated like any other constructor invocation. Define and implement formatting rules for the new syntax. Add formatting tests.
+
+### DDC
+
+If these features can be implemented entirely in the front end with no Kernel
+changes, then no DDC changes may be needed.
+Otherwise, DDC may need to handle any Kernel changes or otherwise add support
+for this.
+
+### IntelliJ
+
+Update the IntelliJ parser to support the new syntax forms.
+
+### Grok
+
+Update Grok to handle the new AST.
+
+### Analysis server
+
+Update to use the latest analyzer with support for the feature. 
+
+The analyzer should likely recognize the grammar even without the experiment flag, and report errors stating that the feature is not available yet, rather than trying to parse `>>>` as `>>` followed by `>`, which is never valid as an operator.
+
+There are no new quick-fixes needed.
+
+### Atom plug-in
+
+The [Dart Atom plug-in][atom] has a grammar for syntax highlighting Dart code in
+Atom. This same grammar is also used for syntax highlighting on GitHub. Update
+this to handle the new syntax.
+
+[atom]: https://github.com/dart-atom/dart
+
+### VS Code
+
+Update the syntax highlighting grammar to support the control flow syntax (and,
+likely apply a very similar diff to the Atom grammar above).
+
+### VM
+
+If the feature is handled by the front end, there may be no VM work. The VM already fully supports `>>>`, but it is possible that generic functions as type arguments may trigger some unused code paths.
+
+### Co19 tests
+
+The co19 team can start implementing tests early using the experimental flags.
+Those tests should not be run by default until the feature has been released.
+
+### Usability validation
+
+No usability testing is planned. There is nothing *new* in these features (deliberately), just allowing existing syntax in new places, or one more operators which also exist in other languages.
+
+## Phase 3 (Release)
+
+### Enabling
+
+The language team updates the experimental flags `generic-metadata` and `triple-shift` to
+always be enabled and no longer be available to users, and releases this update
+in the next stable Dart release.
+
+### Use
+
+The Dart library team introduces `int.operator>>>` as soon as possible.
+
+### Documentation
```

Comment:
@kwalrath I think there's probably some minor documentation work to be done here (e.g. I seem to remember there's a table of operators with their precedence floating around somewhere).

Reply:
Ack. That table should probably be updated.